test/pmi_client.c

```diff
@@ -334,30 +328,6 @@ static int test_item5(void)
 static int test_item6(void)
 {
     int rc = 0;
-    char val[100];
-    const char *tkey = __func__;
-    const char *tval = __FILE__;
-
-    if (PMI_SUCCESS != (rc = PMI_KVS_Put(jobid, tkey, tval))) {
-        log_fatal("PMI_KVS_Put %d\n", rc);
-        return rc;
-    }
-
-    if (PMI_SUCCESS != (rc = PMI_KVS_Get(jobid, tkey, val, sizeof(val)))) {
```

Comment:
I don't think that is actually correct. I can't say for certain about PMI-1, though I believe it is consistent with us, but PMIx clearly requires that I can "put" data and "get" it back for myself without any other action (note this is not true for "get" requests on other procs). At least, that is what we have supported to-date prior to dstore, and I suspect we should retain that behavior as OMPI certainly depends upon it.